core/field.ts

```diff
@@ -45,12 +45,25 @@ import * as WidgetDiv from './widgetdiv.js';
 import type {WorkspaceSvg} from './workspace_svg.js';
 import * as Xml from './xml.js';
 
-export type FieldValidator<T = any> = (value?: T) => T|null|undefined;
+/**
+ * A function that is called to validate changes to the field's value before
+ * they are set.
+ *
+ * **NOTE:** Validation returns one option between `T`, `null`, and `undefined`.
+ *
+ * @see {@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/validators#return_values}
+ * @param newValue The value to be validated.
+ * @returns `T` to set this function's returned value instead of `newValue`.
```

Comment:
Similar to the previous comment, please convert to a single `@returns` tag

Reply:
Fixed!